pkg/liquidity-mining/contracts/admin/AuthorizerAdaptorEntrypoint.sol

```diff
@@ -71,39 +60,32 @@ contract AuthorizerAdaptorEntrypoint is IAuthorizerAdaptorEntrypoint {
-     * adaptor itself.
+     * @notice Returns the action ID associated with calling a given function through the `AuthorizerAdaptor`. Note that
+     * even though the Adaptor's action IDs are not actually used by it (since the Authorizer ignores those values - see
+     * `TimelockAuthorizer.canPerform`), this contract reuses those IDs to simplify migrations and tooling.
      *
-     * @param selector - The 4 byte selector of the function to be called using `performAction`
-     * @return The associated action ID
+     * See `AuthorizerAdaptor.getActionId` for more information on how the action IDs are computed, and how functions
+     * with equal selectors are assigned the same action ID.
      */
     function getActionId(bytes4 selector) public view override returns (bytes32) {
         return getAuthorizerAdaptor().getActionId(selector);
     }
 
-    /**
-     * @notice Performs an arbitrary function call on a target contract, provided the caller is authorized to do so.
-     * @param target - Address of the contract to be called
-     * @param data - Calldata to be sent to the target contract. It should be at least 4 bytes long (i.e. the length of
-     * the selector corresponding to the function to be called)
-     * @return The bytes encoded return value from the performed function call
-     */
     function performAction(address target, bytes calldata data) external payable override returns (bytes memory) {
         // We want to check that the caller is authorized to call the function on the target rather than this function.
         // We must then pull the function selector from `data` rather than `msg.sig`.
-        // Note that if `data` is less than 4 bytes long this will revert.
+
+        // Note that this will revert if `data` is less than 4 bytes long. We test for that to provide a nicer revert
+        // reason.
+        _require(data.length >= 4, Errors.INSUFFICIENT_DATA);
         bytes4 selector = data[0] | (bytes4(data[1]) >> 8) | (bytes4(data[2]) >> 16) | (bytes4(data[3]) >> 24);
 
-        // This call to `canPerform` will validate the actual action ID and sender in the authorizer.
         _require(canPerform(getActionId(selector), msg.sender, target), Errors.SENDER_NOT_ALLOWED);
 
-        // Contracts using the adaptor expect it to be the caller of the actions to perform, so we forward
-        // the call to `performAction` to the adaptor instead of performing it directly.
+        // The `AuthorizerAdaptor` will not check for permissions: it is special-cased in the `TimelockAuthorizer` so
+        // that all calls to it that are not made from this entrypoint fail, while those that originate in the
+        // entrypoint succeed. This works as we have just checked that the caller has permission to perform the action
+        // encoded by `data`. See `TimelockAuthorizer.canPeform` for more details.
```

Comment:
+1.
Shouldn't we clarify why we are not calling the target contract directly at this point? (that was the intent of the deleted comment)

Reply:
Hm no, I don't think so. That's a different topic (_why_ there's an adaptor in the first place).